controls/microsoft-365-foundations-2.1.6.rb

```diff
@@ -0,0 +1,101 @@
+        1. Navigate to Microsoft 365 Defender https://security.microsoft.com.
+        2. Click to expand Email & collaboration select Policies & rules> Threat policies.
+        3. Under Policies select Anti-spam.
+        4. Click on the Anti-spam outbound policy (default).
+        5. Select Edit protection settings then under Notifications
+        6. Check Send a copy of outbound messages that exceed these limits to these users and groups then enter the desired email addresses.
+        7. Check Notify these users and groups if a sender is blocked due to sending outbound spam then enter the desired email addresses.
+        8. Click Save.
+    To set the Exchange Online Spam Policies correctly using PowerShell:
+        1. Connect to Exchange Online using Connect-ExchangeOnline.
+        2. Run the following PowerShell command:
+            $BccEmailAddress = @(\"<INSERT-EMAIL>\")
+            $NotifyEmailAddress = @(\"<INSERT-EMAIL>\")
+            Set-HostedOutboundSpamFilterPolicy -Identity Default -
+            BccSuspiciousOutboundAdditionalRecipients $BccEmailAddress -
+            BccSuspiciousOutboundMail $true -NotifyOutboundSpam $true -
+            NotifyOutboundSpamRecipients $NotifyEmailAddress
+    Note: Audit and Remediation guidance may focus on the Default policy however, if a Custom Policy exists in the organization's tenant then ensure the setting is set as outlined in the highest priority policy listed."
+
+  desc 'rationale',
+       'A blocked account is a good indication that the account in question has been breached and an attacker is using it to send spam emails to other people.'
+
+  impact 0.5
+  tag severity: 'medium'
+  tag cis_controls: [
+    { '8' => ['17.5'] },
+    { '7' => ['7.9'] },
+    { '7' => ['7.10'] }
+  ]
+  tag default_value: "BccSuspiciousOutboundAdditionalRecipients : {}
+                      BccSuspiciousOutboundMail : False
+                      NotifyOutboundSpamRecipients : {}
+                      NotifyOutboundSpam : False"
+  tag nist: ['IR-1', 'IR-8']
+
+  ensure_exchange_online_spam_policies_set_to_notify_admins_script = %{
+    $client_id = '#{input('client_id')}'
+    $certificate_password = '#{input('certificate_password')}'
+    $certificate_path = '#{input('certificate_path')}'
+    $organization = '#{input('organization')}'
+    import-module exchangeonlinemanagement
+    Connect-ExchangeOnline -CertificateFilePath $certificate_path -CertificatePassword (ConvertTo-SecureString -String $certificate_password -AsPlainText -Force)  -AppID $client_id -Organization $organization -ShowBanner:$false
+    Get-HostedOutboundSpamFilterPolicy | Select-Object Name, Bcc*, Notify* | ConvertTo-Json
+  }
+  powershell_output = JSON.parse(powershell(ensure_exchange_online_spam_policies_set_to_notify_admins_script).stdout.strip)
+  case powershell_output
+  when Hash
+    describe "Ensure the following Exchange Online Spam Policy (#{powershell_output['Name']})" do
```

Comment:
This is an area where we don't really need to use RSpec syntax and would probably be better served just using classic InSpec syntax:

``` ruby
describe "Ensure the following Exchange Online Spam Policy (#{powershell_output['Name']})" do
    subject { powershell_output['Name'] }
    its('BccSuspiciousOutboundMail') { should cmp true }
    its('NotifyOutboundSpam') { should cmp true }
    . . .
end
```

etc etc

Right now you're basically rewriting the test output as a string when InSpec's default syntax will give you that for free.